src/devices/Ssd1351/samples/Ssd1351.Sample.cs

```diff
@@ -0,0 +1,105 @@
+            device.SendCommand(new SetMultiplexRatio()); // Use all 128 common lines by default....
+            device.SendCommand(new SetSegmentReMapColorDepth(ColorDepth.ColourDepth65K, CommonSplit.OddEven, Seg0Common.Column127)); //0x74 Color Depth = 64K, Enable COM Split Odd Even, Scan from COM[N-1] to COM0. Where N is the Multiplex ratio., Color sequence is normal: A -> B -> C
+            device.SendCommand(new SetColumnAddress()); // Columns = 0 -> 127
+            device.SendCommand(new SetRowAddress()); // Rows = 0 -> 127
+            device.SendCommand(new SetDisplayStartLine()); // set startline to to 0
+            device.SendCommand(new SetDisplayOffset(0)); // Set vertical scroll by Row to 0-127.
+            device.SendCommand(new SetGpio(GpioMode.Disabled, GpioMode.Disabled)); // Set all GPIO to Input disabled
+            device.SendCommand(new SetVDDSource()); // Enable internal VDD regulator
+            device.SendCommand(new SetPreChargePeriod(2, 3)); // Phase 1 period of 5 DCLKS,  Phase 2 period of 3 DCLKS
+            device.SendCommand(new SetNormalDisplay()); // Reset to Normal Display
+            device.SendCommand(new SetVcomhDeselectLevel()); // 0.82 x VCC
+            device.SendCommand(new SetContrastABC(0xC8, 0x80, 0xC8)); // Contrast A = 200, B = 128, C = 200
+            device.SendCommand(new SetMasterContrast()); // No Change = 15
+            device.SendCommand(new SetVSL()); // External VSL
+            device.SendCommand(new Set3rdPreChargePeriod(0x01)); // Set Second Pre-charge Period = 1 DCLKS
+            device.SendCommand(new SetDisplayOn()); //--turn on oled panel
+            device.ClearScreen();
+        }
+
+        static void Main(string[] args)
+        {
+            const int pinID_DC = 23;
+            const int pinID_Reset = 24;
+
+            SpiDevice displaySPI;
+            GpioController displayGPIO;
+
+            if ((displaySPI = new UnixSpiDevice(new SpiConnectionSettings(0, 0) { Mode = SpiMode.Mode3, DataBitLength = 8, ClockFrequency = 12_000_000 /* 12MHz */ })) != null)
+            {
+                if ((displayGPIO = new GpioController()) != null)
+                {
+
+                    using (Bitmap dotnetBM = new Bitmap(128, 128))
+                    {
+                        using (Ssd1351 ssd1351 = new Ssd1351(displaySPI, displayGPIO, pinID_DC, pinID_Reset))
+                        {
+                            ssd1351.ResetDisplayAsync().Wait();
+                            InitDisplayForAdafruit(ssd1351);
+
+                            using (System.Drawing.Graphics g = System.Drawing.Graphics.FromImage(dotnetBM))
+                            {
+
+                                while(true)
+                                {
+                                    foreach (string filepath in Directory.GetFiles(@"images", "*.png").OrderBy(f => f))
```

Comment:
that's a lot of indentation in here. I'd recommend simplifying this a bit:
- all usings one after another i.e.:

```csharp
using (...)
using (...)
{
   ...
}
```

- `(displayGPIO = new GpioController()) != null` will never be false so if statement is not necessary
- `(displaySPI = new UnixSpiDevice(new SpiConnectionSettings(0, 0) ....) != null` - similarly will never be null

that should reduce most of the nesting

Reply:
Done in the latest PR